In [46]:
from google.colab import drive
import pandas as pd
import numpy as np
import matplotlib as plt

drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Using MMA data to make predictions.

In [47]:
ufc_path = '/content/drive/MyDrive/sports_datasets/ufc_dataset/ufc-master.csv'
mma = pd.read_csv(ufc_path)

print(f'Number of rows in dataframe: {len(mma.index)}')
print(f'Number of columns in dataframe: {len(mma.columns)}')

Number of rows in dataframe: 4850
Number of columns in dataframe: 119


We have sourced the data from https://www.kaggle.com/mdabbert/ultimate-ufc-dataset. Various information about every fight can be displayed. There are 119 columns in total.

In [51]:
# display all coulumns in the data
mma.head()

R_fighter         B_fighter  R_odds  ...  b_sub_odds  r_ko_odds  b_ko_odds
0   Edson Barboza     Giga Chikadze    -125  ...      2500.0      250.0      200.0
1    Bryan Battle    Gilbert Urbina    -200  ...       700.0      350.0      800.0
2   Ricky Turcios    Brady Hiestand    -145  ...       800.0      350.0      800.0
3       Kevin Lee  Daniel Rodriguez    -145  ...      1600.0      700.0      330.0
4  Andre Petroski  Micheal Gillmore    -550  ...      1800.0      225.0      800.0

[5 rows x 119 columns]

In [54]:
# amount of KO's in each weight class
ko_count = mma.groupby('weight_class')[['B_win_by_KO/TKO']].count()
ko_count.rename({'B_win_by_KO/TKO': 'Wins by KO'}, axis=1, inplace=True)
ko_count.sort_values('Wins by KO').reset_index()

ko_count

Wins by KO
weight_class                     
Bantamweight                  487
Catch Weight                   32
Featherweight                 564
Flyweight                     235
Heavyweight                   382
Light Heavyweight             389
Lightweight                   847
Middleweight                  587
Welterweight                  826
Women's Bantamweight          153
Women's Featherweight          17
Women's Flyweight             128
Women's Strawweight           203

In [53]:
def find_fighter_stats(fighter_name):
  if fighter_name in mma:
    return
  else:
    print('')